File: src/api/MixServer.Domain/FileExplorer/Models/Metadata/MediaMetadata.cs
using MixServer.Domain.Tracklists.Dtos.Import;

namespace MixServer.Domain.FileExplorer.Models.Metadata;

public class MediaMetadata(string mimeType, TimeSpan duration, int bitrate, string fileHash, ImportTracklistDto tracklist) : FileMetadata(mimeType), IMediaMetadata
{
    public TimeSpan Duration { get; } = duration;
    public int Bitrate { get; } = bitrate;

    public string FileHash { get; } = fileHash;
    public ImportTracklistDto Tracklist { get; } = tracklist;
}

File: src/clients/mix-server-client/src/app/main-content/file-explorer/models/media-metadata.ts
import {FileMetadata} from "./file-metadata";
import {FormGroup} from "@angular/forms";
import {TracklistForm} from "../../../services/tracklist/models/tracklist-form.interface";
import {TranscodeState} from "../../../generated-clients/mix-server-clients";

export class MediaMetadata extends FileMetadata {
  constructor(mimeType: string,
              public duration: string,
              public bitrate: number,
              public transcodeState: TranscodeState,
              public tracklist: FormGroup<TracklistForm>) {
    super(mimeType);
  }

  override copy() {
    return new MediaMetadata(this.mimeType, this.duration, this.bitrate, this.transcodeState, this.tracklist);
  }
}
